integrations/graal/native-image-extension/src/main/java/io/helidon/integrations/graal/nativeimage/extension/HelidonReflectionFeature.java

```diff
@@ -120,83 +128,135 @@ public void beforeCompilation(BeforeCompilationAccess access) {
         MpConfigProviderResolver.buildTimeEnd();
     }
 
-    @SuppressWarnings("unchecked")
-    private void addAnnotatedWithReflected(BeforeAnalysisContext context) {
-        FeatureImpl.BeforeAnalysisAccessImpl access = context.access();
+    private List<Class<?>> findSubclasses(BeforeAnalysisContext context, String className) {
+        ScanResult scan = context.scan();
+        ClassInfo superclass = scan.getClassInfo(className);
+
+        if (null == superclass) {
+            traceParsing(() -> "Class " + className + " is not on classpath, cannot find subclasses.");
+            return List.of();
+        }
+
+        List<Class<?>> subclasses = scan
+                .getSubclasses(className)
+                .stream()
+                .map(classInfo -> context.access().findClassByName(classInfo.getName()))
+                .filter(Objects::nonNull)
+                .collect(Collectors.toList());
 
+        if (superclass.isInterface()) {
+            List<Class<?>> implementations = scan
+                    .getClassesImplementing(className)
+                    .stream()
+                    .map(classInfo -> context.access().findClassByName(classInfo.getName()))
+                    .filter(Objects::nonNull)
+                    .collect(Collectors.toList());
+
+            List<Class<?>> result = new ArrayList<>(subclasses);
+            result.addAll(implementations);
+            return result;
+        } else {
+            return subclasses;
+        }
+
+    }
+
+    private List<Class<?>> findAnnotated(BeforeAnalysisContext context, String annotation) {
+        return context.scan()
+                .getClassesWithAnnotation(annotation)
+                .stream()
+                .map(classInfo -> {
+                    Class<?> clazz = null;
+                    try {
+                        clazz = classInfo.loadClass();
+                    } catch (Exception e) {
+                        traceParsing(() -> "Class " + classInfo.getName() + " annotated by " + annotation + " cannot be loaded");
+                        if (TRACE_PARSING) {
+                            traceParsing(() -> "\tException class: " + e.getClass().getName() + ", message: " + e.getMessage());
+                        }
```

Comment:
`if (TRACE_PARSING)` is already inside of  `traceParsing`

Reply:
changed from Exception.printStackTrace, forgot to clear up. Will do.

Reply:
Fixed